### PR TITLE
#164046203: Set Up Testing Environment

### PR DESCRIPTION
**Pull request by BrianSerem**
**Description**

This **PR** 
- This sets up the environment for testing.
- Installs Enzyme
- Configures Enzyme inside package.json

<hr>

**Type of change**
- [x] chore(non-breaking change that makes testing possible).
<hr>

**Change log**
App.js file has been moved to components folder
<hr>

**How to test this manually**
- Clone the repository `https://github.com/andela/ah-legion-frontend`
- Checkout the branch `ch-setup-test-environment-164046203`
- Run `npm install` to instal the required necessary node modules
-  Run `npm test a` to run all tests
-  The available tests will run on your console

**Checklist:**
- [x] This follows all the guidelines for this project.

#### Related Stories

[#1164046203](https://www.pivotaltracker.com/story/show/164046203)

### CODE REVIEW

**Comment by MandelaK** [comment]
Please resolve merge conflicts ... also is there any way to get test coverage locally?

**Comment by BrianSerem** [comment]
> Please resolve merge conflicts ... also is there any way to get test coverage locally?

None at the moment, third-party apps should be configured to get test coverage

**Review by ElMonstro — APPROVED** [review]
Your test runs. Approved.

File: src/setupTests.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+configure({ adapter: new Adapter() });
```
Comment (Etomovich) [inline src/setupTests.js:4]: @BrianSerem please add an end of file to this file

File: src/tests/components/App.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    const component = shallow(<App debug />);
+    expect(component.find('div')).toBeDefined();
+  
+  });
+  it('H3 tag displays correctly in "debug" mode', () => {
+    const component = shallow(<App debug />);
+    expect(component.find('div.h3')).toBeDefined();
+  
+  });
+});
```
Comment (Etomovich) [inline src/tests/components/App.test.js:15]: @BrianSerem Please add an end of file to this file too.

Comment (ElMonstro) [reply src/tests/components/App.test.js:15]: @Etomovich the files you have cited EOLs

Comment (BrianSerem) [reply src/tests/components/App.test.js:15]: Etole this file has an end of line, files without have little red arrows at the bottom

File: src/setupTests.js (hunk 1 above)
Comment (BrianSerem) [reply src/setupTests.js:4]: I believe there is an EOL, files without have little red pointer arrows at the bottom.

**Review by MandelaK — APPROVED** [review]

File: src/tests/components/App.test.js (hunk 2 above)
Comment (Etomovich) [reply src/tests/components/App.test.js:15]: This is now ok with me.

File: src/setupTests.js (hunk 1 above)
Comment (Etomovich) [reply src/setupTests.js:4]: Thanks for the clarification @BrianSerem

**Review by Etomovich — APPROVED** [review]

**Review by RachelleMaina — APPROVED** [review]

**Comment by MandelaK** [comment]
I found a fix for the error messages we were getting with @cob04 when running your tests ... Jest needs a dependency on mac an these are the instructions for installing

```
npm r -g watchman
brew install watchman
```

Check out this link for more info
https://github.com/expo/expo/issues/854#issuecomment-343243804

I that does not work, install fsevents as a dev dependency instead like so
```
npm install -s fsevents  
```

Hope this helps

**Review by MandelaK — CHANGES_REQUESTED** [review]
Fix merge conflicts please. I believe you have not rebased the fix for moving redux to production dependencies. Also try and include help message for when we get the dependency error on mac

**Review by cob04 — APPROVED** [review]
Tests run great, good job.

**Comment by BrianSerem** [comment]
> I found a fix for the error messages we were getting with @cob04 when running your tests ... Jest needs a dependency on mac an these are the instructions for installing
> 
> ```
> npm r -g watchman
> brew install watchman
> ```
> Check out this link for more info
> [expo/expo#854 (comment)](https://github.com/expo/expo/issues/854#issuecomment-343243804)
> 
> I that does not work, install fsevents as a dev dependency instead like so
> 
> ```
> npm install -s fsevents  
> ```
> Hope this helps

I believe I don't have to add `fsevents` to dependencies list because it is not required for this PR. Any team member running into errors should follow the above steps or install `x-code  tools`.

**Review by MandelaK — APPROVED** [review]

**Pull request merged**